File: TrabalhoAtleta/TrabalhoAtletaPY/atleta.py
class Atleta:
    def __init__(self, fone, nome, apelido, data_nascimento, pontuacao_acumulada):
        # Atributos do atleta
        self.fone = fone
        self.nome = nome
        self.apelido = apelido
        self.data_nascimento = data_nascimento
        self.pontuacao_acumulada = pontuacao_acumulada

    def __str__(self):
        # Método para representação em string do objeto Atleta
        return f"Nome: {self.nome}, Apelido: {self.apelido}, Telefone: {self.fone}, Data de Nascimento: {self.data_nascimento}, Pontuacao Acumulada: {self.pontuacao_acumulada}"

    def __eq__(self, other):
        # Método para verificar se dois objetos Atleta são iguais
        return self.fone == other.fone

    def __hash__(self):
        # Método para calcular o hash do objeto Atleta
        return hash(self.fone)

File: TrabalhoAtleta/TrabalhoAtletaJAVA/atleta.java
public class Atleta {
    private String fone;
    private String nome;
    private String apelido;
    private String dataNascimento;
    private int pontuacaoAcumulada;

    public Atleta(String fone, String nome, String apelido, String dataNascimento, int pontuacaoAcumulada) {
        this.fone = fone;
        this.nome = nome;
        this.apelido = apelido;
        this.dataNascimento = dataNascimento;
        this.pontuacaoAcumulada = pontuacaoAcumulada;
    }

    // Getters e setters

    @Override
    public String toString() {
        return "Nome: " + nome + ", Apelido: " + apelido + ", Fone: " + fone + ", Data de Nascimento: " + dataNascimento + ", Pontuacao Acumulada: " + pontuacaoAcumulada;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (obj == null || getClass() != obj.getClass()) return false;
        Atleta atleta = (Atleta) obj;
        return fone.equals(atleta.fone);
    }

    @Override
    public int hashCode() {
        return fone.hashCode();
    }
}